doc/api/esm.md

```diff
@@ -681,126 +695,54 @@ of the following:
+ * @param {} context Currently not used
+ * @param {Function} defaultLoad
+ * @returns {Promise<{
+    format: !string,
+    source: !(string | SharedArrayBuffer | Uint8Array),
+  }>}
  */
-export async function getSource(url, context, defaultGetSource) {
-  const { format } = context;
-  if (Math.random() > 0.5) { // Some condition.
-    // For some or all URLs, do some custom logic for retrieving the source.
-    // Always return an object of the form {source: <string|buffer>}.
-    return {
-      source: '...',
-    };
+export async function load(url, context, defaultLoad) {
+  if (path.extname(url) === 'json6') {
+    // If specifying a format, ensure its value is one of the strings in the
+    // preceding table.
+    context.format = 'module';
   }
   // Defer to Node.js for all other URLs.
-  return defaultGetSource(url, context, defaultGetSource);
+  return defaultLoad(url, context, defaultLoad);
 }
 ```
 
-#### `transformSource(source, context, defaultTransformSource)`
-
-> Note: The loaders API is being redesigned. This hook may disappear or its
-> signature may change. Do not rely on the API described below.
-
-* `source` {string|SharedArrayBuffer|Uint8Array}
-* `context` {Object}
-  * `format` {string}
-  * `url` {string}
-* Returns: {Object}
-  * `source` {string|SharedArrayBuffer|Uint8Array}
-
-The `transformSource` hook provides a way to modify the source code of a loaded
-ES module file after the source string has been loaded but before Node.js has
-done anything with it.
-
-If this hook is used to convert unknown-to-Node.js file types into executable
-JavaScript, a resolve hook is also necessary in order to register any
-unknown-to-Node.js file extensions. See the [transpiler loader example][] below.
-
-```js
-/**
- * @param {!(string | SharedArrayBuffer | Uint8Array)} source
- * @param {{
- *   format: string,
- *   url: string,
- * }} context
- * @param {Function} defaultTransformSource
- * @returns {Promise<{ source: !(string | SharedArrayBuffer | Uint8Array) }>}
- */
-export async function transformSource(source, context, defaultTransformSource) {
-  const { url, format } = context;
-  if (Math.random() > 0.5) { // Some condition.
-    // For some or all URLs, do some custom logic for modifying the source.
-    // Always return an object of the form {source: <string|buffer>}.
-    return {
-      source: '...',
-    };
-  }
-  // Defer to Node.js for all other sources.
-  return defaultTransformSource(source, context, defaultTransformSource);
-}
-```
+In a more advanced scenario, this can also be used to transform an unsupported
+source to a supported one (see [Examples](#esm_examples) below).
 
-#### `getGlobalPreloadCode()`
+#### `globalPreload()`
 
-> Note: The loaders API is being redesigned. This hook may disappear or its
-> signature may change. Do not rely on the API described below.
+> Note: In the experimental version, this hook was named
+> `getGlobalPreloadCode()` and has been renamed to `globalPreload()`
```

Comment:
Again, please undo this change. The API is still experimental. We don’t need discussion of previous experimental versions.

Reply:
Would you object to keeping some version of both? I'm fully onboard restoring the original note, but I think there's value in somehow calling out that this was renamed. Perhaps:

```md
> Note: The loaders API is being redesigned. This hook may disappear or its
> signature may change. Do not rely on the API described below.

> Note: This hook was renamed from `getGlobalPreloadCode()` to `globalPreload()`
```

Reply:
Sure, both sounds fine. I would maybe be a little more explanatory in the latter, like “Note: In a previous version of this API, this hook was named `getGlobalPreloadCode`.” This way we can add future changes similarly, like “In a previous version of this API, this hook’s signature was ___.”